src/nvim/option.c

```diff
@@ -3232,6 +3232,34 @@ static int int_cmp(const void *a, const void *b)
   return *(const int *)a - *(const int *)b;
 }
 
+/// Handle setting 'signcolumn' for value 'val'
+///
+/// @return OK when the value is valid, FAIL otherwise
+int check_signcolumn(char_u *val)
+{
+  // check for basic match
+  if (check_opt_strings(val, p_scl_values, false) == OK) {
+    return OK;
+  }
+
+  // check for 'auto:<NUMBER>-<NUMBER>'
+  if (STRLEN(val) == 8
+      && !STRNCMP(val, "auto:", 5)
+      && ascii_isdigit(val[5])
+      && val[6] == '-'
+      && ascii_isdigit(val[7])
```

Comment:
are those checks needed ? the possible values taken by signcolumn is hardcoded so it can't be anything else than a digit.

Reply:
ha sorry I had not seen you had removed  the check using p_scl_values (I think you changed it since I last had a look at this MR). It's fine then.

Reply:
Yup, I've removed it since there seem to be a hard limit on the assertion in the function that check the option values. So, instead of just increase the limit, risking the breaking changes, I simply refactor it to a new one with a more flexible code.